In [98]:
import pandas as pd
df = pd.read_csv('csv-data\\scraping_results.csv')
df.head()

Locality  Zip Code Type of Property Subtype of Property      Price  \
0       Gent      9000        Apartment           apartment   229000.0   
1      Uccle      1180            House               villa  1250000.0   
2  Etterbeek      1040            House               house   875000.0   
3  Elversele      9140            House               villa   825000.0   
4  Kalmthout      2920            House         manor-house  4750000.0   

  Type of Sale  Number of Rooms  Livable Space (m2)  Fully Equipped Kitchen  \
0     for-sale              1.0                  44                       0   
1     for-sale              4.0                 584                       1   
2     for-sale              4.0                 233                       1   
3     for-sale              4.0                 281                       0   
4     for-sale              3.0                 326                       0   

   Furnished  ...  Garden  Garden Area (m2)  Swimming Pool  \
0          0  ...       0               NaN              0   
1          0  ...       1            1085.0              0   
2          0  ...       0               NaN              0   
3          0  ...       0               NaN              1   
4          0  ...       0               NaN              0   

   Surface of the Land (m2)  Number of Facades  Construction Year  PEB  \
0                       NaN                2.0             1918.0    B   
1                    1225.0                4.0             1932.0    G   
2                     112.0                3.0             1929.0    G   
3                    1450.0                4.0             2001.0    B   
4                   34799.0                4.0             2015.0    A   

   Primary Energy Consumption (kWh/m2)  State of the Building  \
0                                190.0                   Good   
1                                402.0            To renovate   
2                                563.0          To be done up   
3                                181.0                 As new   
4                                 95.0                 As new   

                                                 Url  
0  https://www.immoweb.be/en/classified/apartment...  
1  https://www.immoweb.be/en/classified/villa/for...  
2  https://www.immoweb.be/en/classified/house/for...  
3  https://www.immoweb.be/en/classified/villa/for...  
4  https://www.immoweb.be/en/classified/manor-hou...  

[5 rows x 23 columns]

In [99]:
df.shape

(10875, 23)

**How many rows and columns?**


Initially we had a dataset with 23 columns and 10875 inputs.

In [100]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10875 entries, 0 to 10874
Data columns (total 23 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Locality                             10875 non-null  object 
 1   Zip Code                             10875 non-null  int64  
 2   Type of Property                     10875 non-null  object 
 3   Subtype of Property                  10875 non-null  object 
 4   Price                                10851 non-null  float64
 5   Type of Sale                         10875 non-null  object 
 6   Number of Rooms                      10744 non-null  float64
 7   Livable Space (m2)                   10875 non-null  int64  
 8   Fully Equipped Kitchen               10875 non-null  int64  
 9   Furnished                            10875 non-null  int64  
 10  Any Fireplace ?                      10875 non-null  int64  
 11  Terrace                     

We had some inputs that were missing price (24 cases), so we removed cases with 0 price as they would not be useful in further modeling.

In [101]:
df.dropna(subset=['Price'], inplace=True, ignore_index=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10851 entries, 0 to 10850
Data columns (total 23 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Locality                             10851 non-null  object 
 1   Zip Code                             10851 non-null  int64  
 2   Type of Property                     10851 non-null  object 
 3   Subtype of Property                  10851 non-null  object 
 4   Price                                10851 non-null  float64
 5   Type of Sale                         10851 non-null  object 
 6   Number of Rooms                      10720 non-null  float64
 7   Livable Space (m2)                   10851 non-null  int64  
 8   Fully Equipped Kitchen               10851 non-null  int64  
 9   Furnished                            10851 non-null  int64  
 10  Any Fireplace ?                      10851 non-null  int64  
 11  Terrace                     

We had some inputs that were missing number of rooms (131 cases), so we fill these rows with 0.

In [102]:
percentage_of_missing_room_number = round(df['Number of Rooms'].isna().sum() / len(df) * 100, 2)
print(f"Percentage of missing number of rooms: {percentage_of_missing_room_number}%")

Percentage of missing number of rooms: 1.21%


In [103]:

df['Number of Rooms'] = df['Number of Rooms'].fillna(0)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10851 entries, 0 to 10850
Data columns (total 23 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Locality                             10851 non-null  object 
 1   Zip Code                             10851 non-null  int64  
 2   Type of Property                     10851 non-null  object 
 3   Subtype of Property                  10851 non-null  object 
 4   Price                                10851 non-null  float64
 5   Type of Sale                         10851 non-null  object 
 6   Number of Rooms                      10851 non-null  float64
 7   Livable Space (m2)                   10851 non-null  int64  
 8   Fully Equipped Kitchen               10851 non-null  int64  
 9   Furnished                            10851 non-null  int64  
 10  Any Fireplace ?                      10851 non-null  int64  
 11  Terrace                     

We had many missing values ​​for columns like Garden Area, Terrace Area and Surface of the land. So we filled these missing values ​​with 0 as the fact that these areas do not exist.

In [104]:
percentage_of_missing_garden_area = round(df['Garden Area (m2)'].isna().sum() / len(df) * 100, 2)
print(f"Percentage of missing garden area values: {percentage_of_missing_garden_area}%")
#print(f"Percentage of missing garden area values: {percentage_of_missing_garden_area:.2f}%")

percentage_of_missing_terrace_area = round(df['Terrace Area (m2)'].isna().sum() / len(df) * 100, 2)
print(f"Percentage of missing terrace area values: {percentage_of_missing_terrace_area}%")

percentage_of_missing_land_area = round(df['Surface of the Land (m2)'].isna().sum() / len(df) * 100, 2)
print(f"Percentage of missing values for surface of the land: {percentage_of_missing_land_area}%")

Percentage of missing garden area values: 69.58%
Percentage of missing terrace area values: 58.66%
Percentage of missing values for surface of the land: 34.37%


In [105]:
# Replace missing values in the 'Terrace Area (m2)' columns with 0
df['Terrace Area (m2)'] = df['Terrace Area (m2)'].fillna(0)

# Replace missing values in the 'Garden Area (m2)' columns with 0
df['Garden Area (m2)'] = df['Garden Area (m2)'].fillna(0)

# Replace missing values in the 'Surface of the Land (m2)' columns with 0
df['Surface of the Land (m2)'] = df['Surface of the Land (m2)'].fillna(0)

In [106]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10851 entries, 0 to 10850
Data columns (total 23 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Locality                             10851 non-null  object 
 1   Zip Code                             10851 non-null  int64  
 2   Type of Property                     10851 non-null  object 
 3   Subtype of Property                  10851 non-null  object 
 4   Price                                10851 non-null  float64
 5   Type of Sale                         10851 non-null  object 
 6   Number of Rooms                      10851 non-null  float64
 7   Livable Space (m2)                   10851 non-null  int64  
 8   Fully Equipped Kitchen               10851 non-null  int64  
 9   Furnished                            10851 non-null  int64  
 10  Any Fireplace ?                      10851 non-null  int64  
 11  Terrace                     

We had 23.16% missing values ​​for the Number of Facades column. It was decided to modify this data according to the following rule:
1. Replace missing values in the 'Number of Facades' columns with 0
2. Replace 'Number of facades' with 1 where 'Type of Property' is apartment and 'Number of facades' is 0
3. Replace 'Number of Facades' with 2 where 'Subtype of Property' is 'duplex' or 'town-house' and 'Number of Facades' is 0
4. Replace 'Number of Facades' with 4 where 'Type of Property' is House and 'Number of Facades' is 0

In [107]:
percentage_of_missing_number_facades = round(df['Number of Facades'].isna().sum() / len(df) * 100, 2)
print(f"Percentage of missing values for number of facades: {percentage_of_missing_number_facades}%")

Percentage of missing values for number of facades: 23.43%


In [108]:
# Replace missing values in the 'Number of Facades' columns with 0
df['Number of Facades'] = df['Number of Facades'].fillna(0)

# Replace 'Number of facades' with 1 where 'Type of Property' is 0 (apartment) and 'Number of facades' is 0
df.loc[(df['Type of Property'] == 0) & (df['Number of Facades'] == 'Apartment'), 'Number of Facades'] = 1

# Replace 'Number of Facades' with 2 where 'Subtype of Property' is 'duplex' or 'town-house' and 'Number of Facades' is 0
df.loc[(df['Subtype of Property'].isin(['duplex', 'town-house'])) & (df['Number of Facades'] == 0), 'Number of Facades'] = 2

# Replace remaining 'Number of Facades' 0 values for remaining houses with 4
# Replace 'Number of Facades' with 4 where 'Type of Property' is 1 (House) and 'Number of Facades' is 0
df.loc[(df['Type of Property'] == 1) & (df['Number of Facades'] == 'House'), 'Number of Facades'] = 4

In [109]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10851 entries, 0 to 10850
Data columns (total 23 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Locality                             10851 non-null  object 
 1   Zip Code                             10851 non-null  int64  
 2   Type of Property                     10851 non-null  object 
 3   Subtype of Property                  10851 non-null  object 
 4   Price                                10851 non-null  float64
 5   Type of Sale                         10851 non-null  object 
 6   Number of Rooms                      10851 non-null  float64
 7   Livable Space (m2)                   10851 non-null  int64  
 8   Fully Equipped Kitchen               10851 non-null  int64  
 9   Furnished                            10851 non-null  int64  
 10  Any Fireplace ?                      10851 non-null  int64  
 11  Terrace                     

We had 5.74% and 12.13% missing values ​​for the PEB and State of the Building columns respectively, so we replaced these cases by indicating that the information was not specified.

In [110]:
percentage_of_missing_peb = round(df['PEB'].isna().sum() / len(df) * 100, 2)
print(f"Percentage of missing PEB values: {percentage_of_missing_peb}%")

percentage_of_missing_building_state = round(df['State of the Building'].isna().sum() / len(df) * 100, 2)
print(f"Percentage of missing values for state of the building: {percentage_of_missing_building_state}%")

Percentage of missing PEB values: 6.03%
Percentage of missing values for state of the building: 12.28%


In [111]:
df['PEB'] = df['PEB'].fillna('Not specified')
df['State of the Building'] = df['State of the Building'].fillna('Not specified')

In [112]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10851 entries, 0 to 10850
Data columns (total 23 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Locality                             10851 non-null  object 
 1   Zip Code                             10851 non-null  int64  
 2   Type of Property                     10851 non-null  object 
 3   Subtype of Property                  10851 non-null  object 
 4   Price                                10851 non-null  float64
 5   Type of Sale                         10851 non-null  object 
 6   Number of Rooms                      10851 non-null  float64
 7   Livable Space (m2)                   10851 non-null  int64  
 8   Fully Equipped Kitchen               10851 non-null  int64  
 9   Furnished                            10851 non-null  int64  
 10  Any Fireplace ?                      10851 non-null  int64  
 11  Terrace                     

We had 10.62% missing values for the Energy consumption column. It was decided to replace them with the mean value of the entire range.

In [113]:
percentage_of_missing_energy_consumption = round(df['Primary Energy Consumption (kWh/m2)'].isna().sum() / len(df) * 100, 2)
print(f"Percentage of missing values for energy consumption: {percentage_of_missing_energy_consumption}%")

Percentage of missing values for energy consumption: 11.0%


In [114]:
# Replace missing values in the 'Primary Energy Consumption (kWh/m2)' columns with mean value
df['Primary Energy Consumption (kWh/m2)'] = df['Primary Energy Consumption (kWh/m2)'].fillna(round(df['Primary Energy Consumption (kWh/m2)'].mean(),0))

In [115]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10851 entries, 0 to 10850
Data columns (total 23 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Locality                             10851 non-null  object 
 1   Zip Code                             10851 non-null  int64  
 2   Type of Property                     10851 non-null  object 
 3   Subtype of Property                  10851 non-null  object 
 4   Price                                10851 non-null  float64
 5   Type of Sale                         10851 non-null  object 
 6   Number of Rooms                      10851 non-null  float64
 7   Livable Space (m2)                   10851 non-null  int64  
 8   Fully Equipped Kitchen               10851 non-null  int64  
 9   Furnished                            10851 non-null  int64  
 10  Any Fireplace ?                      10851 non-null  int64  
 11  Terrace                     

The Construction Year column was removed because it contained a lot of missing data that would be difficult to restore.


The "Sale Type" column had only one value, "for sale". Therefore, it was not of interest for the analysis and was also removed.

In [116]:
df = df.drop(columns=['Construction Year'])
df = df.drop(columns=['Type of Sale'])

In [117]:
# Find the index of the current 'Any Fireplace ?' column
fireplaces_index = df.columns.get_loc('Any Fireplace ?')

# Create the new 'Open Fire' column based on 'Any Fireplace ?'
df['Open Fire'] = df['Any Fireplace ?'].apply(lambda x: 0 if x == 0 else 1)

# Insert the new 'Open Fire' column at the same index
df.insert(fireplaces_index, 'Open Fire', df.pop('Open Fire'))
df = df.drop(columns=['Any Fireplace ?'])

In [118]:
df.columns

Index(['Locality', 'Zip Code', 'Type of Property', 'Subtype of Property',
       'Price', 'Number of Rooms', 'Livable Space (m2)',
       'Fully Equipped Kitchen', 'Furnished', 'Open Fire', 'Terrace',
       'Terrace Area (m2)', 'Garden', 'Garden Area (m2)', 'Swimming Pool',
       'Surface of the Land (m2)', 'Number of Facades', 'PEB',
       'Primary Energy Consumption (kWh/m2)', 'State of the Building', 'Url'],
      dtype='object')

For the Fully Equipped Kitchen column it was mentioned a value of 1 for Hyper-Equipped or Installed, and 0 for the rest.

It was noticed that some properties were presented on different web pages, so we removed these duplicates by comparing the values ​​for all columns except the URL column.

In [119]:
#Count duplicates for the same properties that have different URL.
df.duplicated(subset=df.columns.difference(['Url'])).value_counts()

False    10701
True       150
Name: count, dtype: int64

In [120]:
#Remove duplicates for the same properties that have different URL.
df.drop_duplicates(subset=df.columns.difference(['Url']), inplace=True)

In [121]:
df = df.drop(columns=['Url'])

In [122]:
df.shape

(10701, 20)

We added a Province column using a function based on the zip (postal) code values.

In [123]:
def add_province(zip):
    first_two_digits = int(str(zip)[:2])
    if 10 <= first_two_digits <= 12:
        return "Brussels-Capital Region"
    elif 13 <= first_two_digits <= 14:
        return "Province of Walloon Brabant"
    elif 15 <= first_two_digits <= 19 or 30 <= first_two_digits <= 34:
        return "Province of Flemish Brabant"
    elif 20 <= first_two_digits <= 29:
        return "Province of Antwerp"
    elif 35 <= first_two_digits <= 39:
        return "Province of Limburg"
    elif 40 <= first_two_digits <= 49:
        return "Province of Liège"
    elif 66 <= first_two_digits <= 69:
        return "Province of Luxembourg"
    elif 50 <= first_two_digits <= 56:
        return "Province of Namur"
    elif 60 <= first_two_digits <= 65 or 70 <= first_two_digits <= 79:
        return "Province of Hainaut"
    elif 80 <= first_two_digits <= 89:
        return "Province of West Flanders"
    elif 90 <= first_two_digits <= 99:
        return "Province of East Flanders"
    else:
        return "Unknown"

df['Province'] = df['Zip Code'].apply(add_province)

zip_code_index = df.columns.get_loc('Zip Code')  # Get the index of 'Zip Code'
df.insert(zip_code_index + 1, 'Province', df.pop('Province'))  # Insert 'Province' at the correct position

In [124]:
df.columns

Index(['Locality', 'Zip Code', 'Province', 'Type of Property',
       'Subtype of Property', 'Price', 'Number of Rooms', 'Livable Space (m2)',
       'Fully Equipped Kitchen', 'Furnished', 'Open Fire', 'Terrace',
       'Terrace Area (m2)', 'Garden', 'Garden Area (m2)', 'Swimming Pool',
       'Surface of the Land (m2)', 'Number of Facades', 'PEB',
       'Primary Energy Consumption (kWh/m2)', 'State of the Building'],
      dtype='object')

In [125]:
df.shape

(10701, 21)

In [126]:
df['Zip Code'] = str(df['Zip Code'])
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10701 entries, 0 to 10850
Data columns (total 21 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   Locality                             10701 non-null  object 
 1   Zip Code                             10701 non-null  object 
 2   Province                             10701 non-null  object 
 3   Type of Property                     10701 non-null  object 
 4   Subtype of Property                  10701 non-null  object 
 5   Price                                10701 non-null  float64
 6   Number of Rooms                      10701 non-null  float64
 7   Livable Space (m2)                   10701 non-null  int64  
 8   Fully Equipped Kitchen               10701 non-null  int64  
 9   Furnished                            10701 non-null  int64  
 10  Open Fire                            10701 non-null  int64  
 11  Terrace                          

So we have:    

7 categorical columns ('Locality', 'Zip Code', 'Province', 'Type of Property', 'Subtype of Property', 'PEB', 'State of the Building'),    

8 numericals columns ('Price', 'Number of Rooms', 'Livable Space (m2)', 'Terrace Area (m2)', 'Garden Area (m2)', 'Surface of the Land (m2)', 'Number of Facades', 'Primary Energy Consumption (kWh/m2)'), 
  
6 boolean columns ('Fully Equipped Kitchen', 'Furnished', 'Open Fire', 'Terrace', 'Garden', 'Swimming Pool').

In [127]:
# Count the number of qualitative and quantitative variables
qualitative_vars = df.select_dtypes(include='object').shape[1]
quantitative_vars = df.select_dtypes(include=['int64', 'float64']).shape[1]
print(f"Qualitative variables: {qualitative_vars}")
print(f"Quantitative variables: {quantitative_vars}")

Qualitative variables: 7
Quantitative variables: 14


In [128]:
df.describe()

Price  Number of Rooms  Livable Space (m2)  \
count  1.070100e+04     10701.000000        10701.000000   
mean   7.471396e+05         3.590973          251.505000   
std    8.492094e+05         2.175236          225.919926   
min    2.750000e+04         0.000000           13.000000   
25%    2.790000e+05         2.000000          111.000000   
50%    5.250000e+05         3.000000          200.000000   
75%    8.500000e+05         4.000000          315.000000   
max    1.050000e+07        40.000000         5596.000000   

       Fully Equipped Kitchen    Furnished     Open Fire       Terrace  \
count            10701.000000  10701.00000  10701.000000  10701.000000   
mean                 0.600037      0.03523      0.090178      0.415475   
std                  0.489913      0.18437      0.286451      0.492827   
min                  0.000000      0.00000      0.000000      0.000000   
25%                  0.000000      0.00000      0.000000      0.000000   
50%                  1.000000      0.00000      0.000000      0.000000   
75%                  1.000000      0.00000      0.000000      1.000000   
max                  1.000000      1.00000      1.000000      1.000000   

       Terrace Area (m2)        Garden  Garden Area (m2)  Swimming Pool  \
count       10701.000000  10701.000000      10701.000000   10701.000000   
mean           15.017568      0.305579        329.392206       0.074853   
std            62.273548      0.460674       1908.529461       0.263166   
min             0.000000      0.000000          0.000000       0.000000   
25%             0.000000      0.000000          0.000000       0.000000   
50%             0.000000      0.000000          0.000000       0.000000   
75%            15.000000      1.000000         60.000000       0.000000   
max          3749.000000      1.000000      95000.000000       1.000000   

       Surface of the Land (m2)  Number of Facades  \
count              10701.000000       10701.000000   
mean                1559.534062           2.295393   
std                 6466.722670           1.475826   
min                    0.000000           0.000000   
25%                    0.000000           2.000000   
50%                  240.000000           2.000000   
75%                 1000.000000           4.000000   
max               160737.000000           9.000000   

       Primary Energy Consumption (kWh/m2)  
count                         10701.000000  
mean                            317.124661  
std                            1390.105435  
min                               1.000000  
25%                             167.000000  
50%                             257.000000  
75%                             334.000000  
max                          100000.000000

In [129]:
df_statistics = df.describe(include = 'all')
print(df_statistics)

         Locality                                           Zip Code  \
count       10701                                              10701   
unique       1376                                                  1   
top     Antwerpen  0        9000\n1        1180\n2        1040\n3...   
freq          429                                              10701   
mean          NaN                                                NaN   
std           NaN                                                NaN   
min           NaN                                                NaN   
25%           NaN                                                NaN   
50%           NaN                                                NaN   
75%           NaN                                                NaN   
max           NaN                                                NaN   

                       Province Type of Property Subtype of Property  \
count                     10701            10701               

We should transform some categorical values as "Province", "Property Type", "PEB", "Building Condition" into quantitive values.

In [130]:
state_mapping = {
    'To restore': 1,
    'To renovate': 2,
    'To be done up': 3,
    'Good': 4,
    'Just renovated': 5,
    'As new': 6,
    'Not specified' : 7 # For 'Not specified' a new group
}

df['State of the Building_Numeric'] = df['State of the Building'].replace(state_mapping)

C:\Users\olyas\AppData\Local\Temp\ipykernel_26340\1136489473.py:11: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['State of the Building_Numeric'] = df['State of the Building'].replace(state_mapping)


In [131]:
PEB_mapping = {
    'Not specified' : 8,
    'G': 7,
    'F': 6,
    'E': 5,
    'D': 4,
    'C': 3,
    'B': 2,
    'A': 1
}

df['PEB_Numeric'] = df['PEB'].replace(PEB_mapping)

C:\Users\olyas\AppData\Local\Temp\ipykernel_26340\1526814067.py:12: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['PEB_Numeric'] = df['PEB'].replace(PEB_mapping)


In [132]:
df['Province'].value_counts()

Province
Brussels-Capital Region        1952
Province of Antwerp            1762
Province of East Flanders      1719
Province of West Flanders      1645
Province of Flemish Brabant    1093
Province of Liège               865
Province of Walloon Brabant     589
Province of Luxembourg          331
Province of Hainaut             325
Province of Namur               244
Province of Limburg             176
Name: count, dtype: int64

In [133]:
province_mapping = {
    'Brussels-Capital Region': 1,
    'Province of Antwerp': 2,
    'Province of East Flanders': 3,
    'Province of West Flanders': 4,
    'Province of Flemish Brabant': 5,
    'Province of Liège': 6,
    'Province of Walloon Brabant': 7,
    'Province of Luxembourg': 8,
    'Province of Hainaut': 9,
    'Province of Namur': 10,
    'Province of Limburg': 11,
}

df['Province_Numeric'] = df['Province'].replace(province_mapping)

C:\Users\olyas\AppData\Local\Temp\ipykernel_26340\3720752789.py:15: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['Province_Numeric'] = df['Province'].replace(province_mapping)


In [134]:
df['Type of Property_Numeric'] = df['Type of Property'].replace({'House': 1, 'Apartment': 0}).infer_objects(copy=False)

C:\Users\olyas\AppData\Local\Temp\ipykernel_26340\4110281198.py:1: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['Type of Property_Numeric'] = df['Type of Property'].replace({'House': 1, 'Apartment': 0}).infer_objects(copy=False)


In [135]:
df['Subtype of Property'].value_counts()

subtype_group_mapping = {
    'house': 'Houses',
    'villa': 'Luxury Properties',
    'town-house': 'Houses',
    'bungalow': 'Houses',
    'farmhouse': 'Houses',
    'country-cottage': 'Houses',
    'chalet': 'Luxury Properties',
    'apartment': 'Apartments',
    'apartment-block': 'Specialized Properties',
    'duplex': 'Apartments',
    'penthouse': 'Apartments',
    'ground-floor': 'Apartments',
    'flat-studio': 'Apartments',
    'triplex': 'Apartments',
    'service-flat': 'Apartments',
    'mansion': 'Luxury Properties',
    'exceptional-property': 'Luxury Properties',
    'castle': 'Luxury Properties',
    'manor-house': 'Luxury Properties',
    'mixed-use-building': 'Specialized Properties',
    'loft': 'Specialized Properties',
    'kot': 'Apartments',
    'other-property': 'Specialized Properties'
}

df['Subtype of Property_Grouped'] = df['Subtype of Property'].map(subtype_group_mapping)

df['Subtype of Property_Grouped'].value_counts()

Subtype of Property_Grouped
Houses                    4622
Apartments                3254
Luxury Properties         2016
Specialized Properties     809
Name: count, dtype: int64

In [136]:
subtype_mapping = {
    'Apartments' : 1,
    'Houses' : 2,
    'Specialized Properties' : 3,
    'Luxury Properties' : 4
}

df['Subtype of Property_Grouped_Numeric'] = df['Subtype of Property_Grouped'].replace(subtype_mapping)

C:\Users\olyas\AppData\Local\Temp\ipykernel_26340\4083140909.py:8: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['Subtype of Property_Grouped_Numeric'] = df['Subtype of Property_Grouped'].replace(subtype_mapping)


In [138]:
df.to_csv('csv-data\\cleaned_dataset_analysis.csv', index=False)